modules/graphdb/templates/02_disk_management.sh.tpl

```diff
@@ -29,10 +29,24 @@ AVAILABLE_VOLUMES=()
 # Function to create a volume
 create_volume() {
   log_with_timestamp "Creating new volume"
+
+# Fetch the output from Terraform
+
+  # Determine the KMS key ARN to use
+  
+  if [ -n "${ebs_external_kms_key}" ]; then
```

Comment:
I still prefer to work with one variable here, when we can bring things down to providing a single variable in the TF. 
This could break with unbound variable easily.